lib/cryptoexchange/services/pairs.rb

```diff
@@ -38,19 +38,32 @@ def fetch_via_api(endpoint = self.class::PAIRS_URL)
         end
       end
 
-      def fetch_via_api_using_post(endpoint = self.class::PAIRS_URL)
-        fetch_response = HTTP.timeout(:write => 2, :connect => 5, :read => 8).post(endpoint)
+      def fetch_via_api_using_post(endpoint = self.class::PAIRS_URL, headers = false, body = false)
+        fetch_response = 
+          if headers && body
+            HTTP.timeout(:write => 2, :connect => 5, :read => 8).headers(headers).post(endpoint, body: body)
+          else
+            HTTP.timeout(:write => 2, :connect => 5, :read => 8).post(endpoint)
```

Comment:
@cySheng the only difference between this method and `fetch_using_post` is to not have `:json => params`?

Reply:
I don't seem to understand what you're question is referring to. The coinut api requires certain params to be sent in the headers and body. If the headers and body is present then attach it to the post request.